dist/pom.xml

```diff
@@ -409,6 +409,42 @@
                     </excludes>
                 </configuration>
             </plugin>
+            <plugin>
+                <!-- The dependency scopes of spark-rapids-jni and jucx are `provided` in aggregator submodule,
+                 this will skip the dedupe processing in `create-parallel-world`.
+                 Unpack spark-rapids-jni and jucx jars to `parallel-world` directory after `create-parallel-world`,
+                 then `parallel-world` will be packaged.
+                 Note all Shims use the same version of spark-rapids-jni and jucx -->
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>unpack-spark-rapids-jni-and-ucx</id>
+                        <!-- run after `create-parallel-world`, the phase equals to the phase of the `create-parallel-world` -->
+                        <phase>generate-resources</phase>
+                        <goals>
+                            <goal>unpack</goal>
+                        </goals>
+                        <configuration>
```

Comment:
To have a more obvious way of reasoning about the order of the dependency processing I would move this unpacks inside ant `build-parallel-worlds.xml`  to `copy-dependencies` target either before or after the for loop.

Reply:
Not found a way to invoke the `unpack` goal of `maven-dependency-plugin` in the `maven-antrun-plugin`.
Just coping the jars in the local `~/.m2` path is not so good,  because we can specify the local repository path when executing `mvn` commands.  The `maven-dependency-plugin` is responsible to copy the right jars.

Now the `unpack` works and the order is right, let's keep it?

Reply:
it can be done similar to the dependency:get code there. 
```
+    <target name="unpack-ucx-and-jni">
+        <resources id="unpackArray">
+            <string>com.nvidia:spark-rapids-jni:${project.version}:jar:${cuda.version}</string>
+            <string>org.openucx:jucx:${ucx.version}</string>
+        </resources>
+        <pathconvert property="unpackList" refid="unpackArray" pathsep=","/>
+        <ac:for list="${unpackList}" param="unpackArtifact" trim="true">
+            <sequential>
+                <echo level="info">dependency:unpack @{unpackArtifact}</echo>
+                <exec executable="${maven.home}/bin/mvn" failonerror="true">
+                    <arg value="dependency:unpack"/>
+                    <arg value="-s" if:set="env.URM_URL"/>
+                    <arg value="${project.basedir}/../jenkins/settings.xml" if:set="env.URM_URL"/>
+                    <arg value="-B"/>
+                    <arg value="-Dmaven.repo.local=${maven.repo.local}" if:set="maven.repo.local"/>
+                    <arg value="-Dartifact=@{unpackArtifact}"/>
+                    <arg value="-DoutputDirectory=${project.build.directory}/parallel-world"/>
+                </exec>
+            </sequential>
+        </ac:for>
+    </target>
``` 

Your suggestion is fine with me.